src/utilities/data-formatter.ts

```diff
@@ -85,7 +96,7 @@ class DataFormatter {
     return result;
   }
 
-  toIndexesInputOutput(value1, value2 = null, maxThreshold = 0) {
+  toIndexesInputOutput(value1: string | number | string[], value2: string | string[] | null = null, maxThreshold = 0) {
```

Comment:
Perhaps this is `Values`?

Reply:
Probably , It depends on the likely change to the values type mentioned in the previous comment, the code might not be able to handle "string[][]" correctly so I left this for later.

Reply:
keeping it as is for now since values now includes string[][] and this function can't handle that.